async-nats/src/jetstream/kv/mod.rs

```diff
@@ -386,6 +386,42 @@ impl Store {
     /// # }
     /// ```
     pub async fn watch<T: AsRef<str>>(&self, key: T) -> Result<Watch<'_>, Error> {
+        self.watch_by_deliver_policy(key, DeliverPolicy::New).await
+    }
+
+    /// Creates a [futures::Stream] over [Entries][Entry]  a given key in the bucket,
+    /// and a [`DeliverPolicy`] enum which yields values whenever there are changes for that key.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), async_nats::Error> {
+    /// use futures::StreamExt;
+    /// let client = async_nats::connect("demo.nats.io:4222").await?;
+    /// let jetstream = async_nats::jetstream::new(client);
+    /// let kv = jetstream.create_key_value(async_nats::jetstream::kv::Config {
+    ///     bucket: "kv".to_string(),
+    ///     history: 10,
+    ///     ..Default::default()
+    /// }).await?;
+    /// let mut entries = kv
+    ///     .watch_by_deliver_policy(
+    ///         "kv",
+    ///         async_nats::jetstream::consumer::DeliverPolicy::LastPerSubject,
+    ///     )
+    ///     .await?;
+    /// while let Some(entry) = entries.next().await {
+    ///     println!("entry: {:?}", entry);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn watch_by_deliver_policy<T: AsRef<str>>(
```

Comment:
I would prefer `watch_with_history`, or... `into_future` builder, as we might have more options later.

Reply:
Yeah I wasn't sure about the name, but now that you mentioned `into_future`, that would be a lot more elegant. I'll focus on that instead.

Reply:
@Jarema just to clarify, would you like me to remove the enum that is being passed as an argument to the function (then default to `LastPerSubject` and rename to `watch_with_history`? Or just to rename it, but keep the enum param (if YES and I do rename it to `watch_with_history` would this always be accurate with the all variants in the `DeliverPolicy`)? Also, had a quick chat with @caspervonb and he's against doing it with a builder pattern.